FirebaseRemoteConfig/SwiftNew/ConfigFetch.swift

```diff
@@ -0,0 +1,814 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import Foundation
+
+#if SWIFT_PACKAGE
+  @_implementationOnly import GoogleUtilities_NSData
+#else
+  import FirebaseInstallations
+  import FirebaseRemoteConfigInterop
+  @_implementationOnly import GoogleUtilities
+#endif // SWIFT_PACKAGE
+
+// TODO(ncooke3): Once Obj-C tests are ported, all `public` access modifers can be removed.
+
+#if RCN_STAGING_SERVER
+  private let serverURLDomain = "https://staging-firebaseremoteconfig.sandbox.googleapis.com"
+#else
+  private let serverURLDomain = "https://firebaseremoteconfig.googleapis.com"
+#endif
+private let serverURLVersion = "/v1"
+private let serverURLProjects = "/projects/"
+private let serverURLNamespaces = "/namespaces/"
+private let serverURLQuery = ":fetch?"
+private let serverURLKey = "key="
+private let requestJSONKeyAppID = "app_id"
+
+private let eTagHeaderName = "Etag"
+
+/// Remote Config Error Info End Time Seconds;
+private let throttledEndTimeInSecondsKey = "error_throttled_end_time_seconds"
+
+/// Fetch identifier for Base Fetch
+private let baseFetchType = "BASE"
+/// Fetch identifier for Realtime Fetch
+private let realtimeFetchType = "REALTIME"
+
+/// HTTP status codes. Ref: https://cloud.google.com/apis/design/errors#error_retries
+private enum FetchResponseStatus: Int {
+  case ok = 200
+  case tooManyRequests = 429
+  case internalError = 500
+  case serviceUnavailable = 503
+  case gatewayTimeout = 504
+}
+
+// MARK: - Used for Testing
```

Comment:
Can these be moved to Tests/Swift/FakeUtils?

Reply:
No, these were just poorly named on my part. They need to be defined in the library module for the dependency injection purposes. I renamed the types in [8a02d59](https://github.com/firebase/firebase-ios-sdk/pull/14286/commits/8a02d59196df4bcd0649b89ba145961988622896)